jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/GKEClusterRule/EnableNodeAutoUpgrade.java

```diff
@@ -0,0 +1,120 @@
+            mustFilter.put(PacmanRuleConstants.LATEST, true);
+
+            try {
+                isKeyEnabled = checkKeyEnabled(vmEsURL, mustFilter,key);
+                if (!isKeyEnabled) {
+                    List<LinkedHashMap<String, Object>> issueList = new ArrayList<>();
+                    LinkedHashMap<String, Object> issue = new LinkedHashMap<>();
+
+                    annotation = Annotation.buildAnnotation(ruleParam, Annotation.Type.ISSUE);
+                    annotation.put(PacmanSdkConstants.DESCRIPTION, description);
+                    annotation.put(PacmanRuleConstants.SEVERITY, severity);
+                    annotation.put(PacmanRuleConstants.CATEGORY, category);
+                    issue.put(PacmanRuleConstants.VIOLATION_REASON, violationReason);
+                    issueList.add(issue);
+                    annotation.put("issueDetails", issueList.toString());
+                    logger.debug("========rule ended with status failure {}", annotation);
+                    return new RuleResult(PacmanSdkConstants.STATUS_FAILURE, PacmanRuleConstants.FAILURE_MESSAGE,
+                            annotation);
+                }
+
+            } catch (Exception exception) {
+                throw new RuleExecutionFailedExeption(exception.getMessage());
+            }
+        }
+
+        logger.debug("======== ended with status true=========");
+        return new RuleResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+    }
+    private boolean checkKeyEnabled(String vmEsURL, Map<String, Object> mustFilter,String key) throws Exception {
+        logger.debug("========verifyports  started=========");
+        JsonArray hitsJsonArray = GCPUtils.getHitsArrayFromEs(vmEsURL, mustFilter);
+        boolean validationResult = true;
+        if (hitsJsonArray.size() > 0) {
+            logger.debug("========verifyports hit array=========");
+
+            JsonObject gkeCluster = (JsonObject) ((JsonObject) hitsJsonArray.get(0))
+                    .get(PacmanRuleConstants.SOURCE);
+
+            logger.debug("Validating the data item: {}", gkeCluster.toString());
+
+            JsonArray nodePools = null;
+            if(gkeCluster.getAsJsonObject().get(PacmanRuleConstants.NODE_POOLS)!=null){
+                nodePools=gkeCluster.getAsJsonObject().get(PacmanRuleConstants.NODE_POOLS).getAsJsonArray();
+                for (JsonElement nodePool:nodePools){
+                    logger.info("auto upgrade {}",nodePool.getAsJsonObject());
+                    if(nodePool.getAsJsonObject().get(key)!=null &&nodePool.getAsJsonObject().get(key).getAsBoolean()==false){
+                        validationResult=false;
+                    }
+                }
+            }
+
+        }
+
+        return validationResult;
+    }
+    @Override
+    public String getHelpText() {
+        return "Enable  GKE nodes management properties";
+    }
```

Comment:
Some of the changes are already added with #609, please rebase once to master

Reply:
Done